Polyfills/Window/Source/Window.cpp

```diff
@@ -103,11 +148,31 @@ namespace
         }
 
     private:
+        TimeoutId NextTimeoutId()
+        {
+            while (true)
+            {
+                ++m_lastTimeoutId;
+
+                if (m_lastTimeoutId <= 0)
+                {
+                    m_lastTimeoutId = 1;
+                }
+
+                if (m_idMap.find(m_lastTimeoutId) == m_idMap.end())
+                {
+                    return m_lastTimeoutId;
+                }
+            }
+        }
+
         Babylon::JsRuntime& m_runtime;
         std::thread m_thread;
         std::mutex m_mutex{};
         std::condition_variable m_condVariable{};
-        std::priority_queue<TimeoutFunction, std::vector<TimeoutFunction>, decltype(&TimeoutFunction::Compare)> m_queue{TimeoutFunction::Compare};
+        TimeoutId m_lastTimeoutId = 0;
+        std::map<TimeoutId, Timeout> m_idMap;
+        std::map<Milliseconds, std::list<Timeout>> m_timeMap;
```

Comment:
These maps are weird to me. There should only be one owner of the `Timeout` struct. I think it would be better if the id map is the owner of the `Timeout` struct and the second container is a sorted list of pointers to the Timeout struct in the id map. I suppose vice versa should also work (i.e., sorted list is the owner and id map points to the entry in the sorted list).

Reply:
Would you use bare pointers for this? ...or is there an STL pointer type that would be better?

Reply:
Raw pointers would be fine for this.